aiidalab_widgets_base/structures.py

```diff
@@ -1,88 +1,138 @@
+            description='Or choose from examples:',
+            style={'description_width': '160px'},
+        )
+        supported_formats = ipw.HTML("""All supported structure formats are listed
+        <a href="https://wiki.fysik.dtu.dk/ase/_modules/ase/io/formats.html" target="_blank">here</a>""")
+
         self.viewer = nglview.NGLWidget()
         self.btn_store = ipw.Button(
             description='Store in AiiDA', disabled=True)
         self.structure_description = ipw.Text(
             placeholder="Description (optional)")
 
         self.structure_ase = None
-        self.structure_node = None
+        self._structure_node = None
         self.data_format = ipw.RadioButtons(
             options=self.DATA_FORMATS, description='Data type:')
 
-        if node_class is None:
-            store = ipw.HBox(
-                [self.btn_store, self.data_format, self.structure_description])
-        elif node_class not in self.DATA_FORMATS:
-            raise ValueError("Unknown data format '{}'. Options: {}".format(
-                node_class, self.DATA_FORMATS))
+        if examples:
+            get_structure = ipw.HBox([self.file_upload, self.select_example])
         else:
-            self.data_format.value = node_class
-            store = ipw.HBox([self.btn_store, self.structure_description])
-
-        children = [self.file_upload, self.viewer, store]
-
+            get_structure = self.file_upload
+
+        if storable:
+            if node_class is None:
+                store = [self.btn_store, self.data_format, self.structure_description]
+            elif node_class not in self.DATA_FORMATS:
+                raise ValueError("Unknown data format '{}'. Options: {}".format(
+                    node_class, self.DATA_FORMATS))
+            else:
+                self.data_format.value = node_class
+                store = [self.btn_store, self.structure_description]
+        else:
+            store = [self.structure_description]
+        store = ipw.HBox(store)
+        children = [get_structure, supported_formats, self.viewer, store]
         super(StructureUploadWidget, self).__init__(
             children=children, **kwargs)
 
         self.file_upload.observe(self._on_file_upload, names='data')
+        self.select_example.observe(self._on_select_example, names=['value'])
         self.btn_store.on_click(self._on_click_store)
 
+    @staticmethod
+    def get_example_structures(examples):
+        if examples:
+            to_return=[("Select structure", False)]
+            to_return += [(s.split('/')[-1], s) for s in examples]
+            return to_return
+        else:
+            return []
+
     # pylint: disable=unused-argument
     def _on_file_upload(self, change):
-        self.tmp_folder = tempfile.mkdtemp()
-        tmp = self.tmp_folder + '/' + self.file_upload.filename
-        with open(tmp, 'w') as f:
+        self.file_path = tempfile.mkdtemp() + '/' + self.file_upload.filename
```

Comment:
can we transform this into another `with` (one for the folder, one for the file) so that we are sure both are deleted?

Reply:
ah hm... the file is still needed lateron right?

Reply:
> ah hm... the file is still needed lateron right?

yes, right
A not for myself: replace it with `path.join()`

Reply:
done